### PR TITLE
Implement R4RepositorySubjectProvider logic for managingOrganization and partOf.

### PR DESCRIPTION
**Pull request by lukedegruchy**
- Implement logic for Organization-based subjectIds, including managingOrganization and optionally partOf
- Introduce new configuration for partOf:  SubjectProviderOptions, captured by MeasureEvaluationOptions, to contain the partOf configuration, which is a boolean
- New test class

Closes https://github.com/cqframework/clinical-reasoning/issues/587

### CODE REVIEW

File: cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/SubjectProviderOptions.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+package org.opencds.cqf.fhir.cr.measure;
+
+/**
+ * All options that are pertinent to {@link org.opencds.cqf.fhir.cr.measure.common.SubjectProvider}
+ * implementations, such as {@link org.opencds.cqf.fhir.cr.measure.r4.R4RepositorySubjectProvider}.
+ */
+public class SubjectProviderOptions {
+
+    private boolean isPartOfEnabled;
```
Comment (Capt-Mac) [inline cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/measure/SubjectProviderOptions.java:9]: should have a javaDoc or comment explaining what functionality this option enables

File: cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/MultiMeasureServiceTest.java [hunk 2]
```diff
@@ -931,4 +931,22 @@ void MultiMeasure_EightMeasures_ReporterNotAcceptedResource() {
                 .hasSubjectReference("Practitioner/tester")
                 .hasReporter("Patient/male-2022"));
     }
+
+    @Test
+    void MultiMeasure_EightMeasures_SubjectOrganization() {
+        var when = GIVEN_REPO
+                .when()
```
Comment (Capt-Mac) [inline cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/MultiMeasureServiceTest.java:938]: Question: Is this test notEnabling partOf?
Suggestion: add comment on what functionality is being validated

File: cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/R4RepositorySubjectProviderTest.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                        List.of(resourcify(ResourceType.Organization, ORG_ID_1)),
+                        Stream.of(PAT_ID_1)
+                                .map(id -> resourcify(ResourceType.Patient, id))
+                                .collect(Collectors.toList())));
+    }
+
+    @ParameterizedTest
+    @MethodSource("getSubjectsParams")
+    void getSubjects(
+            R4RepositorySubjectProvider testSubject,
```
Comment (Capt-Mac) [inline cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/R4RepositorySubjectProviderTest.java:90]: Nice test! We should add the other potential subject provider options here too.

File: cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/R4RepositorySubjectProviderTest.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+                        TEST_SUBJECT_ENABLE_PART_OF,
+                        MeasureEvalType.SUBJECT,
+                        List.of(resourcify(ResourceType.Organization, ORG_ID_1)),
+                        // TODO:  LD:  this is technically incorrect:  it should be PAT_ID_1, PAT_ID_2
+                        // However, due to the fact that both InMemoryFhirRepository and IgRepository
+                        // do NOT support chained searches, the results can only be accurately verified
+                        // with a repository that supports DAOs.
+                        Stream.of(PAT_ID_1, PAT_ID_1)
+                                .map(id -> resourcify(ResourceType.Patient, id))
+                                .collect(Collectors.toList())),
```
Comment (Capt-Mac) [inline cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/R4RepositorySubjectProviderTest.java:77]: suggestion: Now that we are Java 17 compliant, we can replace this usage of 'Stream.collect(Collectors.toList())' with 'Stream.toList()' and ensure that the list is unmodified.

File: cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/R4RepositorySubjectProviderTest.java [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+                Arguments.of(
+                        TEST_SUBJECT_DISABLE_PART_OF,
+                        MeasureEvalType.SUBJECT,
+                        List.of(resourcify(ResourceType.Organization, ORG_ID_1)),
+                        Stream.of(PAT_ID_1)
+                                .map(id -> resourcify(ResourceType.Patient, id))
+                                .collect(Collectors.toList())));
```
Comment (Capt-Mac) [inline cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/R4RepositorySubjectProviderTest.java:84]: suggestion: Now that we are Java 17 compliant, we can replace this usage of 'Stream.collect(Collectors.toList())' with 'Stream.toList()' and ensure that the list is unmodified.

File: cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/R4RepositorySubjectProviderTest.java [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+    @ParameterizedTest
+    @MethodSource("getSubjectsParams")
+    void getSubjects(
+            R4RepositorySubjectProvider testSubject,
+            MeasureEvalType measureEvalType,
+            List<String> subjectIds,
+            List<String> expectedSubjects) {
+        final List<String> actualSubjects =
+                testSubject.getSubjects(repository, measureEvalType, subjectIds).collect(Collectors.toList());
```
Comment (Capt-Mac) [inline cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/R4RepositorySubjectProviderTest.java:95]: suggestion: Now that we are Java 17 compliant, we can replace this usage of 'Stream.collect(Collectors.toList())' with 'Stream.toList()' and ensure that the list is unmodified.

File: cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/R4RepositorySubjectProviderTest.java (hunk 3 above)
Comment (lukedegruchy) [reply cqf-fhir-cr/src/test/java/org/opencds/cqf/fhir/cr/measure/r4/R4RepositorySubjectProviderTest.java:187]: Done

**Review by Capt-Mac — APPROVED** [review]

**Pull request merged**